flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/GraphCsvReader.java

```diff
@@ -0,0 +1,471 @@
+	 * @param  type2 The type of CSV field 2 for the CSV reader used for reading Edge data and the type of Edge Value in the returned Graph.
+	 * @return The {@link org.apache.flink.graph.Graph} with Edges and Vertices extracted from the parsed CSV data.
+	 */
+	public Graph<K,VV,EV> types(Class<K> type0, Class<VV> type1, Class<EV> type2) {
+		DataSet<Tuple3<K, K, EV>> edges = this.EdgeReader.types(type0, type0, type2);
+		if(vertexPath!=null) {
+			DataSet<Tuple2<K, VV>> vertices = this.VertexReader.types(type0, type1);
+			return Graph.fromTupleDataSet(vertices, edges, executionContext);
+		} else {
+			return Graph.fromTupleDataSet(edges, this.mapper, executionContext);
+		}
+
+
+	}
+	/**
+	 * Specifies the types for the Graph fields and returns a Graph with those field types in the special case
+	 * where Vertices don't have a value
+	 * @param type0 The type of CSV field 0 and 1 for the CSV reader used for reading Edge data and the type of Vetex ID in the returned Graph.
+	 * @param  type1 The type of CSV field 2 for the CSV reader used for reading Edge data and the type of Edge Value in the returned Graph.
+	 * @return The {@link org.apache.flink.graph.Graph} with Edge extracted from the parsed CSV data and Vertices mapped from Edges with null Value.
+	 */
+	public Graph<K, NullValue, EV> typesVertexValueNull(Class<K> type0, Class<EV> type1) {
+		DataSet<Tuple3<K, K, EV>> edges = this.EdgeReader.types(type0, type0, type1);
+		return Graph.fromTupleDataSet(edges, executionContext);
+	}
+
+	/**
+	 * Specifies the types for the Graph fields and returns a Graph with those field types ain the special case
+	 * where Edges don't have a value
+	 * @param type0 The type of CSV field 0 and 1 for the CSV reader used for reading Edge data and the type of Vetex ID in the returned Graph.
+	 * @param  type1 The type of CSV field 2 for the CSV reader used for reading Edge data and the type of Edge Value in the returned Graph.
+	 * @return The {@link org.apache.flink.graph.Graph} with Edge extracted from the parsed CSV data and Vertices mapped from Edges with null Value.
+	 */
+	public Graph<K, VV, NullValue> typesEdgeValueNull(Class<K> type0, Class<VV> type1) {
+		DataSet<Tuple3<K, K, NullValue>> edges = this.EdgeReader.types(type0, type0)
+				.map(new MapFunction<Tuple2<K, K>, Tuple3<K, K, NullValue>>() {
+					@Override
+					public Tuple3<K, K, NullValue> map(Tuple2<K, K> tuple2) throws Exception {
+						return new Tuple3<K, K, NullValue>(tuple2.f0, tuple2.f1, NullValue.getInstance());
+					}
+				});
+
+		if(vertexPath!=null) {
+			DataSet<Tuple2<K, VV>> vertices = this.VertexReader.types(type0, type1);
+			return Graph.fromTupleDataSet(vertices, edges, executionContext);
+		} else {
+			return Graph.fromTupleDataSet(edges, this.mapper, executionContext);
+		}
+	}
+
+	/**
+	 * Specifies the types for the Graph fields and returns a Graph with those field types.
+	 * This method is overloaded for the case in which Vertices and Edges don't have a value
```

Comment:
Same here.. Actually, are you allowing reading edges with no values? It seems no when reading the docs, but it appears so here :)